src/test/clojure/xtdb/stagnant_log_flusher_test.clj

```diff
@@ -0,0 +1,174 @@
+(ns xtdb.stagnant-log-flusher-test
+  (:require [clojure.test :as t]
+            [xtdb.api :as xt]
+            [xtdb.log :as log]
+            [xtdb.node :as node]
+            [xtdb.test-util :as tu]
+            [xtdb.util :as util]
+            [xtdb.vector.reader :as ivr])
+  (:import (java.io Closeable)
+           (java.util.concurrent Semaphore)
+           (org.apache.arrow.memory BufferAllocator)
+           (org.apache.arrow.vector.ipc ArrowStreamReader)
+           (xtdb.indexer Indexer)
+           (xtdb.log Log)
+           (xtdb.object_store ObjectStore)))
+
+(defonce log-level :error)
+
+(comment
+  (def log-level :debug)
+  (def log-level :info)
+  (def log-level :error)
+  )
+
+(defmacro spin-until
+  [ms expr]
+  `(loop [ret# ~expr
+          wait-until# (+ ~ms (System/currentTimeMillis))]
+     (cond
+       ret# ret#
+       (< wait-until# (System/currentTimeMillis)) ret#
+       :else (recur ~expr wait-until#))))
+
+(defmacro spin-ensure [ms expr]
+  `(loop [ret# ~expr
+          wait-until# (+ ~ms (System/currentTimeMillis))]
+     (cond
+       (not ret#) ret#
+       (< wait-until# (System/currentTimeMillis)) ret#
+       :else (recur ~expr wait-until#))))
+
+(def ^:dynamic *spin-ms*
+  "Change if tolerances change and tests need more time (such as slower CI machines), used for `spin`."
+  500)
+
+(defmacro spin [expr] `(spin-until *spin-ms* ~expr))
+
+(defn each-fixture [f]
+  (let [log-nses '[xtdb.stagnant-log-flusher
+                   xtdb.indexer]
+        old-levels (mapv tu/get-log-level! log-nses)]
+    (try
```

Comment:
if `util/with-log-level` isn't suitable here, maybe we could figure out what we can do to make it suitable?

Reply:
I did at one point have more nses in the list, believe me it was actually uglier! Now it is 2 tis probably ok for `with`as it stands. If we want to make it work nicely for `n` levels we can make it take multiple levels. What about pair style: 

```clojure 
(tu/with-log-levels
  [xtdb.stagnant-log-flusher :info 
   xtdb.indexer :debug]
  (do-stuff))
```

Not sure what you think?
  
Again I try to not change shared utils unless essential as part of an unrelated PR, but if you are happy I can change it. (I would argue for separate issue, refactor task)

Reply:
yep, pair style would work for me. :+1:

Reply:
we could also add a fixture, on reflection:

```clojure
(defn with-log-level [ns level f]
  ...)

(t/use-fixtures :each (tu/with-log-level ns1 l1) (tu/with-log-level ns2 l2) your-each-fixture)
```